lib/dry/types/schema.rb

```diff
@@ -284,6 +284,19 @@ def lax
         Lax.new(schema(keys.map(&:lax)))
       end
 
+      # Merges given schema keys into current schema.
+      #
+      # A new instance is returned.
+      #
+      # @param schema [Schema]
+      # @return [Schema]
+      #
+      # @api public
+      def merge(schema)
+        with(keys: keys + schema.keys)
```

Comment:
repeated names should be filtered here. Otherwise, you can end up with `keys` having duplicates, this is not good.